controllers/watcher_handler.go

```diff
@@ -0,0 +1,96 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023 Datadog, Inc.
+
+package controllers
+
+import (
+	"fmt"
+
+	chaosv1beta1 "github.com/DataDog/chaos-controller/api/v1beta1"
+	"github.com/DataDog/chaos-controller/watchers"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+type newWatcher func(name string, r *DisruptionReconciler, instance *chaosv1beta1.Disruption) (watchers.Watcher, error)
+
+var watcherList = map[string]newWatcher{
+	"chaos-pod": func(name string, r *DisruptionReconciler, instance *chaosv1beta1.Disruption) (watchers.Watcher, error) {
+		return watchers.NewChaosPodWatcher(name, r.Recorder, instance, r.log)
+	},
+	"disruption-target": func(name string, r *DisruptionReconciler, instance *chaosv1beta1.Disruption) (watchers.Watcher, error) {
+		return watchers.NewDisruptionTargetWatcher(name, r.Recorder, r.Reader, r.EnableObserver, instance, r.MetricsSink, r.log)
+	},
+}
+
+// manageInstanceWatchers manage the watchers for each disruption instance
+func (r *DisruptionReconciler) manageInstanceWatchers(instance *chaosv1beta1.Disruption) error {
+	r.log.Debugf("start to manage watchers for the %s disruption", instance.Name)
+
+	if len(watcherList) == 0 {
+		return nil
+	}
+
+	if err := r.WatcherManager.RemoveExpiredWatchers(); err != nil {
+		return err
+	}
+
+	for wName, newWatcherFunc := range watcherList {
+		watcherName, err := getInstanceWatcherName(wName, instance)
+		if err != nil {
+			return fmt.Errorf("failed to initialize %s watcher of the disruption %s: could not get the watcher name. Error: %s", wName, instance.Name, err.Error())
```

Comment:
I believe that errors should always be interpolated with `%w`, https://go.dev/blog/go1.13-errors
```suggestion
			return fmt.Errorf("failed to initialize %s watcher of the disruption %s: could not get the watcher name. Error: %w", wName, instance.Name, err)
```

Reply:
Thank you for the source! Really interesting, I understand now why it is interesting to use `%w` 👍

Reply:
After reading the article and in particular this sentence :
<img width="837" alt="image" src="https://user-images.githubusercontent.com/8859832/230606774-3f8f659a-6f2e-43d4-8810-356765224d92.png">

Should I have to use the `%w` into the watcher implementation? @ptnapoleon 

IMO: the watcher is an abstraction layer, so if I understand the article it is not recommended to wrap error.

Reply:
Given this is all internal to us, and the logs are all read by us, I think it's good to pierce the abstraction layer within the logs

Reply:
It makes sense! I will adapt all the logs 👍